src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java

```diff
@@ -73,14 +73,21 @@ public void setWorker(final CitizenData citizen)
         // If we have a worker, it no longer works here
         if (worker != null)
         {
+            EntityCitizen tempCitizen = worker.getCitizenEntity();
             worker.setWorkBuilding(null);
+            tempCitizen.setLastJob(getJobName());
         }
 
         worker = citizen;
 
         // If we set a worker, inform it of such
         if (worker != null)
         {
+            EntityCitizen tempCitizen = citizen.getCitizenEntity();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'tempCitizen' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)